### PR TITLE
Add page about step conditionals

### PR DESCRIPTION
**Pull request by harrietgrace**
This PR adds a page about the new step conditionals, and adds it to the Pipelines section of the sidebar.

### CODE REVIEW

File: pages/pipelines/conditionals.md.erb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+steps:
+	- command: ./scripts/tests.sh
+	  label: tests
+	  - if:
+	  	build.message != "skip me"
+```
+
+- use it in any step type
+- any of the supported expressions and parameters
+- cannot be used at the same time as the `branch` parameter
```
Comment (keithpitt) [inline pages/pipelines/conditionals.md.erb]: I think it's documented as `branches`

File: pages/pipelines/conditionals.md.erb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+- use the `if` param to conditionally run a step
+- the below example will only run the `tests` step if the build message doesn't contain the string "skip me"
+
+```
+steps:
+	- command: ./scripts/tests.sh
+	  label: tests
+	  - if:
+	  	build.message != "skip me"
+```
```
Comment (keithpitt) [inline pages/pipelines/conditionals.md.erb]: This should be:

```
steps:
    - command: ./scripts/tests.sh
      label: tests
      if: build.message != "skip me"
```

Or for multiple lines:

```
steps:
    - command: ./scripts/tests.sh
      label: tests
      if: |
        // this is a comment
        build.message != "skip me"
```

File: pages/pipelines/conditionals.md.erb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+- use it in any step type
+- any of the supported expressions and parameters
+- cannot be used at the same time as the `branch` parameter
+
+## Supported Expressions
+
+ - Comparators: == != =~ !~
```
Comment (keithpitt) [inline pages/pipelines/conditionals.md.erb]: In some languages, regular expressions can be on either side of an operator, but in our language, they can only be on the right hand side. i.e. this is totally valid ruby:

```ruby
irb(main):006:0> "hello" =~ /hello/
=> 0
irb(main):007:0> /hello/ =~ "hello"
=> 0
```

But for us, you can only do write it like this:

```
"hello" =~ /hello/
```

File: pages/pipelines/conditionals.md.erb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+ - Logical operators: || &&
+ - Integers 12345
+ - Strings 'foobar' or "foobar"
+ - Booleans true false
+ - Parenthesis to control order of evaluation ( )
+ - Regular expressions /^v1\.0/
+ - Function calls foo("bar")
```
Comment (keithpitt) [inline pages/pipelines/conditionals.md.erb]: I don't think we'll have any functions built-in for launch, so not sure if we should mention this in our docs.

**Comment by keithpitt** [comment]
@harrietgrace omg! You've done heaps already! Looks great so far! Since you've laid it all out a couple of extra things jumped into my head, so I added notes on them.

File: pages/pipelines/conditionals.md.erb (hunk 4 above)
Comment (harrietgrace) [reply pages/pipelines/conditionals.md.erb]: Oop, I removed it from the example code but forgot it for this bit 😄

**Comment by harrietgrace** [comment]
@matthewd could you please have a quick look over this? It's here in the preview bizzo: https://buildkite-docs-review-pr-530.herokuapp.com/docs/pipelines/conditionals

Let me know if there's anything you think would be good to add in before we launch it 😊Do you know if there's anything else that needs to happen for conditionals? Is it feature flagged or is it already out there for people to use?

**Comment by matthewd** [comment]
> @matthewd could you please have a quick look over this?

Sorry I missed this @harrietgrace! 🤦🏻‍♂️

**Comment by harrietgrace** [comment]
@matthewd no worries at all! It wasn't urgent 😊

**Pull request merged**